internal/v5/api_test.go

```diff
@@ -2935,6 +2935,21 @@ var publishErrorsTests = []struct {
 		Message: `cannot publish to "unpublished"`,
 		Code:    params.ErrBadRequest,
 	},
+}, {
+	about:  "unknown resource provided",
+	method: "PUT",
+	id:     "~who/trusty/wordpress-0",
+	body: mustMarshalJSON(params.PublishRequest{
+		Resources: map[string]int{
+			"unknown": 5,
+		},
+		Channels: []params.Channel{params.StableChannel},
+	}),
+	expectStatus: http.StatusBadRequest,
+	expectBody: params.Error{
+		Message: `bad request: charm published with incorrect resources: charm does not have resource "unknown"`,
+		Code:    params.ErrBadRequest,
+	},
```

Comment:
I'd like to see the test where we have "known1, unknown, known2" resources and after failure, a check which resource is uploaded/available.